### DISCUSSION TITLE
Newbie Questions About The Homebrew Installation Script, Related To M1 Macs

### DISCUSSION THREAD

**Discussion by rdfuhr** [Homebrew/discussions#411] [Getting started]
I have an M1 MacBook Pro, and I took a look at the current version (as of December 31, 2020) of the Homebrew installation script, which is located at https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh

I have not run this script, and have the following questions.  Bear with me, since I am not fluent in this scripting language. 

1. In this snippet of the script, the intention is to set HOMEBREW_PREFIX to "/opt/homebrew for arm64 Macs, but what is the meaning and the intent of the code that says _if [[ -z "${HOMEBREW_ON_LINUX-}" ]]; then_... in the following?  Is the script to be interpreted that we will only execute the code to set HOMEBREW_PREFIX="/opt/homebrew" if HOMEBREW_ON_LINUX is false (and also of course that "$UNAME_MACHINE"=="arm64"?

if [[ -z "${HOMEBREW_ON_LINUX-}" ]]; then
  if [[ "$UNAME_MACHINE" == "arm64" ]]; then
    # On ARM macOS, this script installs to /opt/homebrew only
    HOMEBREW_PREFIX="/opt/homebrew"

2. Does the user need to create the directory /opt/homebrew before running the script?

**Comment by carlocab** [comment]
>what is the meaning and the intent of the code

You'll want to read `man bash`. In particular, you are looking for the sections on `CONDITIONAL EXPRESSIONS` and `EXPANSION`. You can also read about the equivalents for Zsh in `man zshmisc` and `man zshexpn`.

>Does the user need to create the directory /opt/homebrew

No. The script will tell you everything you need to do when you run it. It also tells you everything it does as it runs.
